File: backend/src/main/java/turip/content/controller/dto/response/ContentPlaceResponse.java
package turip.content.controller.dto.response;


import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalTime;
import turip.content.domain.ContentPlace;
import turip.place.controller.dto.response.PlaceResponse;

public record ContentPlaceResponse(
        Long id,
        int visitDay,
        int visitOrder,
        PlaceResponse place,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "mm:ss")
        LocalTime timeLine,
        boolean isFavoritePlace
) {

    public static ContentPlaceResponse of(ContentPlace contentPlace, boolean isFavoritePlace) {
        return new ContentPlaceResponse(
                contentPlace.getId(),
                contentPlace.getVisitDay(),
                contentPlace.getVisitOrder(),
                PlaceResponse.from(contentPlace.getPlace()),
                contentPlace.getTimeLine(),
                isFavoritePlace
        );
    }
}

File: mobile/core/data/src/commonMain/kotlin/com/on/turip/core/data/dto/content/ContentPlaceResponse.kt
package com.on.turip.core.data.dto.content

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ContentPlaceResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("place")
    val place: PlaceResponse,
    @SerialName("visitDay")
    val visitDay: Int,
    @SerialName("visitOrder")
    val visitOrder: Int,
    @SerialName("timeLine")
    val timeLine: String,
    @SerialName("isTuripPlace")
    val isTuripPlace: Boolean,
)
